esp-hal/src/i2c.rs

```diff
@@ -73,6 +75,13 @@ cfg_if::cfg_if! {
     }
 }
 
+// Chunk writes/reads by this size
+#[cfg(any(esp32, esp32s2))]
+const I2C_CHUNK_SIZE: usize = 32;
+
+#[cfg(not(any(esp32, esp32s2)))]
```

Comment:
technically not necessary but would otherwise be a subtle behavioral change for chips other than ESP32/ESP32-S2